app/code/Magento/Persistent/README.md

```diff
@@ -1,5 +1,63 @@
-Magento\Persistent module enables set customer a long-term cookie containing internal id (random hash - to exclude brute
+# Magento_Persistent module
+
+This module enables set customer a long-term cookie containing internal id (random hash - to exclude brute
 force) of persistent session. Persistent session data is kept in DB - so it's not deleted in some days and is kept for
 as much time as we need. DB session keeps customerId + some data from real customer session that we want to sync (e.g.
 num items in shopping cart). For registered customer this info is synced to persistent session if choose "Remember me"
 checkbox during first login.
+
+## Installation
+
+Before installing this module, note that the Magento_Persistent is dependent on the following modules:
+- `Magento_Checkout`
+- `Magento_PageCache`
+
+The Magento_Persistent module creates the `persistent_session` table in the database.
+
+This module modifies the following tables in the database:
+- `quote` - adds column `is_persistent`
+
+For information about a module installation in Magento 2, see [Enable or disable modules](https://devdocs.magento.com/guides/v2.4/install-gde/install/cli/install-cli-subcommands-enable.html).
+
+## Extensibility
+
+Extension developers can interact with the Magento_Persistent module. For more information about the Magento extension mechanism, see [Magento plug-ins](https://devdocs.magento.com/guides/v2.4/extension-dev-guide/plugins.html).
+
+[The Magento dependency injection mechanism](https://devdocs.magento.com/guides/v2.4/extension-dev-guide/depend-inj.html) enables you to override the functionality of the Magento_Persistent module.
+
+A lot of functionality in the module is on JavaScript, use [mixins](https://devdocs.magento.com/guides/v2.4/javascript-dev-guide/javascript/js_mixins.html) to extend it.
+
+### Events
+
+The module dispatches the following events:
+
+#### Controller
+
+- `persistent_session_expired` event in the `\Magento\Persistent\Controller\Index\UnsetCookie::execute` method
+
+#### Observer
+
+- `persistent_session_expired` event in the `\Magento\Persistent\Observer\CheckExpirePersistentQuoteObserver::execute` method
+
+For information about an event in Magento 2, see [Events and observers](http://devdocs.magento.com/guides/v2.4/extension-dev-guide/events-and-observers.html#events).
+
+### Layouts
+
+This module introduces the following layouts in the `view/frontend/layout` directory:
```

Comment:
Actually it doesn't introduce those - it's only a "client" of layout handles introduced by Magneto_Customer - please remove them from here.
I initially haven't noticed that this comment should be added here, but better late than never.